attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/model/AttributeMaskingRequest.java

```diff
@@ -101,6 +102,7 @@ public LeafResource getResource() {
     }
 
     @Generated
+    @SuppressWarnings({"java:S3740", "rawtypes"}) // Provide the parametrized type for this generic
     public Rules getRules() throws JsonProcessingException {
         return MAPPER.treeToValue(this.rules, Rules.class);
```

Comment:
```suggestion
    @Generated
    // Generic wildcard types should not be used in return types
    // But we don't necessarily know the record datatype
    @SuppressWarnings({"java:S1452"})
    public Rules<?> getRules() throws JsonProcessingException {
        return MAPPER.treeToValue(this.rules, Rules.class);
```
Would this be possible/more consistent?

n.b. I believe `S3740` is equivalent to `rawtypes` but looks like there's an issue in SQ where it complains about both, thus why we can go from two to one

Reply:
They are equivalent so I have reduced the warning to 1 value with the description. I think the "rawtypes" is a more generic rule that shows in IntelliJ but is not an issue within SQ.

Reply:
With this updated the smell "Provide the parametrized type for this generic." is being thrown by SQ which is ignored with the "rawtypes" warning